### PR TITLE
Poetry install command replaced & poetry check command added

### PR DESCRIPTION
**Pull request by sanjeevz3009**
# Description

Currently, we aren't checking if the poetry lock file matches the environment when using poetry/ when using poetry to install packages in the sml-catalogue repo.

If you have not checked any of the lists below explain why here.

## Type of change

- Replaced `poetry install` command with the `poetry install --sync` to ensure it matches the lock file
- `poetry check` command was added to validate the content of the pyproject.toml file and its consistency with the poetry.lock file. It returns a detailed report if there are any errors.

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Technical Enhancements

# Checklist:

If any of these are not completed, please explain why in the notes.

## **Definition of Done**
**Code and merges**

- [x] Code to be commented on where applicable 
- [x] Documentation updated where required 
- [x] Have considered non-functional requirements such as Security, Performance, Scalability, and Fault Tolerance
- [x] I have linted the code

**Testing**

- [x] All levels of acceptance test are passing (automated, integration, manual, accessibility, etc.)
- [x] I have run the behave command to check the selenium behaviour tests pass locally
- [x] Acceptance criteria met

**Other Checks**
- [x] I have performed a self-review of my own code
- [x] I have checked for spelling errors
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes don't break anything unexpected
- [x] I have checked and updated the security.txt file where required
- [x] Up to date with the main branch

### CODE REVIEW

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2uicqnnwijrw1.cloudfront.net/
cloudfront_id=E3JINSIUUTQ61C

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2uicqnnwijrw1.cloudfront.net/
cloudfront_id=E3JINSIUUTQ61C

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2uicqnnwijrw1.cloudfront.net/
cloudfront_id=E3JINSIUUTQ61C

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2uicqnnwijrw1.cloudfront.net/
cloudfront_id=E3JINSIUUTQ61C

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2uicqnnwijrw1.cloudfront.net/
cloudfront_id=E3JINSIUUTQ61C

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2uicqnnwijrw1.cloudfront.net/
cloudfront_id=E3JINSIUUTQ61C

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2uicqnnwijrw1.cloudfront.net/
cloudfront_id=E3JINSIUUTQ61C

File: ci/tasks/build_and_deploy/build_files.sh [hunk 1]
```diff
@@ -3,7 +3,6 @@
 # Runs static code testing and builds files
 : "${BUILD_TYPE}"
 : "${RELEASE_CANDIDATE}"
-: "${SIGNING_KEY}"
```
Comment (gibbardsteve) [inline ci/tasks/build_and_deploy/build_files.sh:6]: Why has this been removed?

File: ci/tasks/build_and_deploy/build_files.sh [hunk 2]
```diff
@@ -28,9 +29,6 @@ run_linting(){
 }
 git config --global user.email "spp-shared-services@example.com"
 git config --global user.name "spp-shared-services"
-git config --global user.signingkey "${SIGNING_KEY}"
```
Comment (gibbardsteve) [inline ci/tasks/build_and_deploy/build_files.sh:31]: This has been removed?

File: pyproject.toml [hunk 3]
```diff
@@ -3,7 +3,7 @@ name = "sml-catalogue"
 version = "0.1.1-rc.1"
 description = "This repo contains the build environment and code to generate and upload the Statistical Methods Library (SML) Portal Web Application."
 authors = []
-readme = "README.md"
```
Comment (gibbardsteve) [inline pyproject.toml:6]: It looks like your changes are undo-ing previously merged changes?

**Review by gibbardsteve — COMMENTED** [review]
A couple of comments about potential merge issues

**Comment by sanjeevz3009** [comment]
Closing branch due to merging issues!

**Pull request closed without merging**